Make Joplin resizable

<!--
  Please provide a clear and concise description of what the bug is. (In the section Steps To Reproduce.)
  Include screenshots if needed.
  Please test using the latest Joplin release to make sure your issue has not already been fixed.
-->

<!--
  IMPORTANT: If you are reporting a clipper bug, please include an example URL that shows the issue.
  Without the URL the issue is likely to be closed.
-->

## Environment

Joplin version: `1.0.333`
Platform: Android 10
OS specifics: it's an Android Desktop

## Steps to reproduce

1. Start Joplin
2. Try to resize it

## Describe what you expected to happen

Joplin should be resizable.

I'm using [Samsung DeX](https://www.samsung.com/us/explore/dex/), which gives a desktop like experience and makes use of Android windowing support.

[Samsung has documentation on how to enable window resizing for Android applications](https://developer.samsung.com/samsung-dex/modify-optimizing.html).
In most cases this just requires to modify the [`AndroidManifest.xml`](https://github.com/laurent22/joplin/blob/dev/ReactNativeClient/android/app/src/main/AndroidManifest.xml):

```xml
<application
    android:resizeableActivity="true">
</application>
```

[GitNex also added ](https://codeberg.org/gitnex/GitNex/pulls/601/files) `android:configChanges` and two other meta tags.

This documentation also has more tweaks and information on how to improve the desktop experience in case you're interested.
They also have [an app testing guide](https://developer.samsung.com/samsung-dex/testing.html) (although some apps are fine with this guide but won't work on "real" DeX sometimes).

At the moment the resizing button is greyed out and the application is always nearly squarish:

![Screenshot of Joplin on Samsung DeX](https://user-images.githubusercontent.com/372620/88278136-30b40200-cce2-11ea-84f7-6c29584dd3e8.jpg)

**PS:** I know that it is possible to _force_ all apps to be resizable via `Samsung Labs` > `Force Resize` but it would obviously better to make the application natively resizable to create less friction.
**PPS:** I'm not sure whether this is relevant but [this](https://codeberg.org/gitnex/GitNex/pulls/601/files) is how GitNex fixed the issue.

@laurent22 is there any documentation how to build android packages?

I tried

```bash
$ ANDROID_SDK_ROOT=~/Android/Sdk/ node Tools/release-android.js --type=release
```




<details>
  <summary>But it failed (click to expand)</summary>

```
Updating version numbers in build.gradle...
Creating release: 1.3.12
Running from: /projects/joplin
Building APK file v1.3.12...
./gradlew assembleRelease -PbuildDir=build
Fatal error
Error: Command failed: ./gradlew assembleRelease -PbuildDir=build
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: /projects/joplin/ReactNativeClient/node_modules/react-native-camera/android/src/main/java/com/google/android/cameraview/Camera2.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: /projects/joplin/ReactNativeClient/node_modules/@react-native-community/geolocation/android/src/main/java/com/reactnativecommunity/geolocation/GeolocationModule.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: /projects/joplin/ReactNativeClient/node_modules/@react-native-community/slider/android/src/main/java/com/reactnativecommunity/slider/ReactSliderManager.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: /projects/joplin/ReactNativeClient/node_modules/@react-native-community/slider/android/src/main/java/com/reactnativecommunity/slider/ReactSliderManager.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
Note: /projects/joplin/ReactNativeClient/node_modules/react-native-fs/android/src/main/java/com/rnfs/RNFSManager.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: /projects/joplin/ReactNativeClient/node_modules/react-native-share/android/src/main/java/cl/json/RNSharePathUtil.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: /projects/joplin/ReactNativeClient/node_modules/react-native-version-info/android/src/main/java/cx/evermeet/versioninfo/RNVersionInfoModule.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: /projects/joplin/ReactNativeClient/node_modules/react-native-webview/android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
error While resolving module `lib/rnInjectedJs/webviewLib`, the Haste package `lib` was found. However the module `rnInjectedJs/webviewLib` could not be found within the package. Indeed, none of these files exist:

  * `/projects/joplin/ReactNativeClient/lib/rnInjectedJs/webviewLib(.native|.android.js|.native.js|.js|.android.json|.native.json|.json|.android.ts|.native.ts|.ts|.android.tsx|.native.tsx|.tsx)`
  * `/projects/joplin/ReactNativeClient/lib/rnInjectedJs/webviewLib/index(.native|.android.js|.native.js|.js|.android.json|.native.json|.json|.android.ts|.native.ts|.ts|.android.tsx|.native.tsx|.tsx)`. Run CLI with --verbose flag for more details.

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:bundleReleaseJsAndAssets'.
> Process 'command 'node'' finished with non-zero exit value 1

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 2m 20s

    at ChildProcess.exithandler (node:child_process:309:12)
    at ChildProcess.emit (node:events:327:20)
    at maybeClose (node:internal/child_process:1048:16)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:288:5) {
  killed: false,
  code: 1,
  signal: null,
  cmd: './gradlew assembleRelease -PbuildDir=build'
}

```

</details>

I did `npm install` before but it still complains about missing `webviewLib`.

--- a/packages/app-mobile/android/app/src/main/AndroidManifest.xml
+++ b/packages/app-mobile/android/app/src/main/AndroidManifest.xml
@@ -34,6 +34,7 @@
 		android:allowBackup="true"
 		android:networkSecurityConfig="@xml/network_security_config"
 		android:requestLegacyExternalStorage="true"
+		android:resizeableActivity="true"
 		android:theme="@style/AppTheme">
 
 		<!-- RN-NOTIFICATION -->

--- a/packages/app-mobile/android/app/src/main/AndroidManifest.xml
+++ b/packages/app-mobile/android/app/src/main/AndroidManifest.xml
@@ -25,6 +25,9 @@
 
 		android:allowBackup: used to enable Android Backup which some users need:
 		https://github.com/laurent22/joplin/issues/4020
+
+		android:resizeableActivity: allows user to resize Joplin window
+		https://github.com/laurent22/joplin/issues/3547
 	-->
 	<application
 		android:name=".MainApplication"

